Signal/src/ViewControllers/ConversationView/ConversationViewItem.m

```diff
@@ -430,6 +470,16 @@ - (void)ensureViewState:(YapDatabaseReadTransaction *)transaction
         self.messageCellType = OWSMessageCellType_TextMessage;
         self.displayableBodyText = [[DisplayableText alloc] initWithFullText:@"" displayText:@"" isTextTruncated:NO];
     }
+
+    if (message.quotedMessage && message.quotedMessage.authorId.length > 0
+        && (message.quotedMessage.body.length > 0 || message.quotedMessage.contentType.length > 0)) {
```

Comment:
Given we have a quotedMessage, when is message.quotedMessage.authorId == 0? Is that an error?

Also, is it an error if neither body nor attachment is set?

Reply:
I've already done a bunch of this work, but I've just made a card to audit this later.